### PR TITLE
Testing and Mocking updates and corrections

### PR DESCRIPTION
**Pull request by milesfrain**
Really excellent articles!

Here's a summary of the changes:
* Continue to use minimal example for tips section after setup.
* Show code coverage command and output for minimal example.
* Add link to CppUMockGen.
* Use tagged version of littlefs for compatability.
* Match minimal src to snippet.
* Fix failing complex example.

### CODE REVIEW

File: _posts/2019-10-08-unit-testing-basics.md [hunk 1]
```diff
@@ -1066,7 +1066,7 @@ $ brew install cpputest
 On Ubuntu, it can be installed using `apt`:
 
 ```
-$ sudo apt-get install cpputest
+$ sudo apt install cpputest
```
Comment (milesfrain) [inline _posts/2019-10-08-unit-testing-basics.md:1069]: Seems like convention these days is to go with the shorter form.

File: _posts/2019-10-08-unit-testing-basics.md [hunk 2]
```diff
@@ -1117,15 +1117,15 @@ Most unit test frameworks will generate separate binaries for each `.cpp` unit
 test file written so that you can load them in a debugger (lldb or gdb).
 
 ```
-$ lldb build/kv_store/kv_store_tests
-(lldb) target create "build/kv_store/kv_store_tests"
-Current executable set to 'build/kv_store/kv_store_tests' (x86_64).
+$ lldb build/sum/sum_tests
```
Comment (milesfrain) [inline _posts/2019-10-08-unit-testing-basics.md:1120]: More straightforward for users continuing from minimal example

File: _posts/2019-10-08-unit-testing-basics.md [hunk 3]
```diff
@@ -1138,9 +1138,39 @@ tests, so you can be sure that the piece of code was tested in some capacity.
 Note that code coverage doesn't measure the different behaviors a code path
 **could take**, but only that a particular code path **was taken**.
 
-![]({% img_url unit-testing-basics/code_coverage.png %})
+```
+$ sudo apt install lcov
+$ make lcov
```
Comment (milesfrain) [inline _posts/2019-10-08-unit-testing-basics.md:1143]: Wanted to promote that this already setup in the makefile

File: _posts/2019-10-08-unit-testing-basics.md [hunk 4]
```diff
@@ -1138,9 +1138,39 @@ tests, so you can be sure that the piece of code was tested in some capacity.
...
-Above is a an example coverage report taken from the Memfault Public SDK[^5].
+...
+
+genhtml -o test_coverage -t "coverage" --num-spaces 4 build/lcov.info -o build/test_coverage/
+Reading data file build/lcov.info
+Found 1 entries.
+Writing .css and .png files.
+Generating output.
+Processing file minimal/my_sum.c
+Writing directory view page.
```
Comment (milesfrain) [inline _posts/2019-10-08-unit-testing-basics.md:1161]: could strip out a few more of these lines to make this snippet less verbose

File: example/unit-testing/README.md [hunk 5]
```diff
@@ -38,7 +40,9 @@ The tests can be run by running:
 
 ```
 $ cd complex
-$ git clone https://github.com/ARMmbed/littlefs.git
+$ git clone -b v2.1.2 https://github.com/ARMmbed/littlefs.git
```
Comment (milesfrain) [inline example/unit-testing/README.md:43]: `emubd` was renamed to `bd` in [`v2.2.0`](https://github.com/ARMmbed/littlefs/tree/v2.2.0), but wanted to stick with the exact version (by date) that I assume you used for the article.

File: example/unit-testing/complex/tests/src/test_protocol.cpp [hunk 6]
```diff
@@ -54,20 +54,19 @@ TEST(TestProtocolParser, Hello) {
   };
 
   mock()
-      .expectOneCall("a")
```
Comment (milesfrain) [inline example/unit-testing/complex/tests/src/test_protocol.cpp:57]: Not sure if this was an intentional failure for demonstration purposes, but I think it's better to let readers introduce their own failures.

File: example/unit-testing/minimal/tests/src/test_my_sum.cpp [hunk 7]
```diff
@@ -13,6 +13,5 @@ TEST_GROUP(TestMySum) {
 };
 
 TEST(TestMySum, Test_MySumBasic) {
-  int rv = my_sum(3, 4);
-  LONGS_EQUAL(7, rv);
+  LONGS_EQUAL(7, my_sum(3, 4));
```
Comment (milesfrain) [inline example/unit-testing/minimal/tests/src/test_my_sum.cpp:16]: now matches the code snippet in the article

File: _posts/2020-05-12-unit-test-mocking.md [hunk 8]
```diff
@@ -205,11 +205,12 @@ mocks using Ruby, CppUMock mocks are just normal functions with some extra
...
 
 When using CppUMock, if you have a few functions you are trying to mock out,
-it's probably best to keep them defined in test file, especially if you think
+it's probably best to keep them defined in the test file, especially if you think
 it's an isolated case. If you are mocking out a large number of functions, or
 think it might be used by other teammates or in future tests, go ahead and move
-it to its own file. Mocks can be easily generated for CppUMock using one of two
-tools, [mockify](https://github.com/marco-m/mockify) or
+it to its own file. Mocks can be easily generated for CppUMock using any of the
+following tools: [CppUMockGen](https://github.com/jgonzalezdr/CppUMockGen),
```
Comment (milesfrain) [inline _posts/2020-05-12-unit-test-mocking.md:212]: added [CppUMockGen](https://github.com/jgonzalezdr/CppUMockGen) to the list

File: _posts/2019-10-08-unit-testing-basics.md (hunk 2 above)
Comment (tyhoff) [reply _posts/2019-10-08-unit-testing-basics.md:1120]: <3

File: example/unit-testing/README.md (hunk 5 above)
Comment (tyhoff) [reply example/unit-testing/README.md:43]: perfect, great idea

File: _posts/2019-10-08-unit-testing-basics.md (hunk 4 above)
Comment (tyhoff) [reply _posts/2019-10-08-unit-testing-basics.md:1161]: Agreed, proposing the change

File: _posts/2019-10-08-unit-testing-basics.md [hunk 9]
```diff
@@ -1138,9 +1138,39 @@ tests, so you can be sure that the piece of code was tested in some capacity.
...
+Overall coverage rate:
+  lines......: 100.0% (2 of 2 lines)
+    functions..: 100.0% (1 of 1 function)
+
+$ firefox build/test_coverage/index.html
+```
```
Comment (tyhoff) [inline _posts/2019-10-08-unit-testing-basics.md:1167]:
```suggestion
Overall coverage rate:
  lines......: 100.0% (2 of 2 lines)
    functions..: 100.0% (1 of 1 function)
```

File: _posts/2019-10-08-unit-testing-basics.md (hunk 3 above)
Comment (tyhoff) [reply _posts/2019-10-08-unit-testing-basics.md:1143]: <3

**Review by tyhoff — APPROVED** [review]
Took me until a few hours later to realize who this is! 👋 

Really appreciate you taking the time to update the post.

**Pull request merged**